db/trie/mptrie/branchnode.go

```diff
@@ -80,8 +80,10 @@ func (b *branchNode) MarkAsRoot() {
 func (b *branchNode) Children() []node {
 	trieMtc.WithLabelValues("branchNode", "children").Inc()
 	children := []node{}
-	for _, child := range b.children {
-		children = append(children, child)
+	for index := 0; index < radix; index++ {
```

Comment:
we better understand why it works in old code (before mptrie refactor)

the fix should make it work by same logic as before, otherwise this might be a breaking change